modules/apps/item-selector/item-selector-taglib/src/main/resources/META-INF/resources/atoms/imageSelectorCoverImageAtom.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {State} from '@liferay/frontend-js-state-web';
+
+const imageSelectorCoverImageAtom = State.atom<{
+	imageData: {
+		attributeDataImageId?: string;
+		coverImageFileEntryId?: string;
+		entryId?: string;
+		fileEntryId?: string;
+		groupId?: string;
+		mimeType?: string;
+		randomId?: string;
+		title?: string;
+		type?: string;
+		url?: string;
+		uuid?: string;
+	};
```

Comment:
This is a somewhat loose type based on what I saw at runtime and what I could see being generated on the server. In reality, no existing listener accesses any of these fields, so we could probably simplify this a bit (at least, make it match the type passed in to `setImage()`, which is what I am talking about in that other TODO).

FWIW, this is the sole TS file in `image-selector-taglib`. Adding it just to do The Right Thing™ from the get-go and have type info for all atoms we create.

Reply:
I see what you mean... maybe @ambrinchaudhary wants to weigh in here seeing as she'll take this over when she moves back to lima 😉

Reply:
Let me push an update to this in a few minutes that makes this simpler and we can discuss that.